code/modules/power/singularity/emitter.dm

```diff
@@ -308,3 +354,149 @@
+
+
+/obj/machinery/power/emitter/prototype
+	name = "Prototype Emitter"
+	icon = 'icons/obj/turrets.dmi'
+	icon_state = "protoemitter"
+	icon_state_on = "protoemitter_+a"
+	can_buckle = TRUE
+	buckle_lying = 0
+	var/view_range = 12
+	var/datum/riding/riding_datum = null
+
+//BUCKLE HOOKS
+
+/obj/machinery/power/emitter/prototype/unbuckle_mob(mob/living/buckled_mob,force = 0)
+	playsound(src,'sound/mecha/mechmove01.ogg', 50, 1)
+	manual = FALSE
+	for(var/obj/item/I in buckled_mob.held_items)
+		if(istype(I, /obj/item/weapon/turret_control))
+			qdel(I)
+	if(istype(buckled_mob))
+		buckled_mob.pixel_x = 0
+		buckled_mob.pixel_y = 0
+		if(buckled_mob.client)
+			buckled_mob.client.change_view(world.view)
+	for(var/datum/action/innate/protoemitter/H in buckled_mob.actions)
+		qdel(H)
+	. = ..()
+
+/obj/machinery/power/emitter/prototype/user_buckle_mob(mob/living/M, mob/living/carbon/user)
+	if(user.incapacitated() || !istype(user))
+		return
+	for(var/atom/movable/A in get_turf(src))
+		if(A.density)
+			if(A != src && A != M)
+				return
+	M.loc = get_turf(src)
+	..()
+	playsound(src,'sound/mecha/mechmove01.ogg', 50, 1)
+	M.pixel_y = 14
+	layer = 4.1
+	if(M.client)
+		M.client.change_view(view_range)
+	var/datum/action/innate/protoemitter/firing/auto = new()
+	auto.Grant(M, src)
+
+/datum/action/innate/protoemitter
+	check_flags = AB_CHECK_RESTRAINED | AB_CHECK_STUNNED | AB_CHECK_CONSCIOUS
+	var/obj/machinery/power/emitter/prototype/PE
+	var/mob/living/carbon/U
+
+
+/datum/action/innate/protoemitter/Grant(mob/living/L, obj/machinery/power/emitter/prototype/proto)
+	PE = proto
+	U = L
+	..()
+
+/datum/action/innate/protoemitter/firing
+	name = "Switch to Manual Firing"
+	desc = "The emitter will only fire on your command and at your designated target"
+	button_icon_state = "mech_zoom_on"
+
+/datum/action/innate/protoemitter/firing/Activate()
+	if(PE.manual == TRUE)
+		playsound(PE,'sound/mecha/mechmove01.ogg', 50, 1)
+		PE.manual = FALSE
+		name = "Switch to Manual Firing"
+		desc = "The emitter will only fire on your command and at your designated target"
+		button_icon_state = "mech_zoom_on"
+		for(var/obj/item/I in U.held_items)
+			if(istype(I, /obj/item/weapon/turret_control))
+				qdel(I)
+		UpdateButtonIcon()
+		return
+	if(PE.manual == FALSE)
+		playsound(PE,'sound/mecha/mechmove01.ogg', 50, 1)
+		name = "Switch to Automatic Firing"
+		desc = "Emitters will switch to periodic firing at your last target"
+		button_icon_state = "mech_zoom_off"
+		PE.manual = TRUE
+		for(var/V in U.held_items)
+			var/obj/item/I = V
+			if(istype(I))
+				if(U.dropItemToGround(I))
+					var/obj/item/weapon/turret_control/TC = new /obj/item/weapon/turret_control()
+					U.put_in_hands(TC)
+			else	//Entries in the list should only ever be items or null, so if it's not an item, we can assume it's an empty hand
+				var/obj/item/weapon/turret_control/TC = new /obj/item/weapon/turret_control()

```

Comment:
So long as you accept that this is what the code is doing, and you're fine with it, while weird I don't particularly have any issues with it, you're essentially saying the emitter is powerful enough that it needs balancing so that it takes all your hands to use, which is fine.